cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/PartitionedFileSet.java

```diff
@@ -49,6 +49,24 @@
   void addPartition(PartitionKey key, String path);
 
   /**
+   * Add a partition for a given partition key, stored at a given path (relative to the file set's base path),
+   * with the given metadata.
+   */
+  void addPartition(PartitionKey key, String path, Map<String, String> metadataEntries);
```

Comment:
Name the parameter `metadata` to be consistent with the `PartitionOuput.setMetadata`.